app/routes/circulars/circulars.server.ts

```diff
@@ -0,0 +1,212 @@
+/*!
+ * Copyright © 2022 United States Government as represented by the Administrator
+ * of the National Aeronautics and Space Administration. No copyright is claimed
+ * in the United States under Title 17, U.S. Code. All Other Rights Reserved.
+ *
+ * SPDX-License-Identifier: NASA-1.3
+ */
+
+import { tables } from '@architect/functions'
+import type { DynamoDBDocument } from '@aws-sdk/lib-dynamodb'
+import { dynamoDBAutoIncrement } from '~/lib/dynamoDBAutoIncrement'
+import { getUser } from '../__auth/user.server'
+
+export type CircularModel = {
+  circularId?: number
+  sub?: string
+  createdOn: number
+  subject: string
+  body: string
+  submitter: string
+}
+
+export class CircularsServer {
+  #sub?: string
+  #email?: string
+  #groups?: string[]
+  #name?: string
+  #affiliation?: string
+  #isLocal: boolean
+
+  private constructor(
+    isLocal: boolean,
+    sub?: string,
+    email?: string,
+    groups?: string[],
+    name?: string,
+    affiliation?: string
+  ) {
+    this.#affiliation = affiliation
+    this.#email = email
+    this.#groups = groups
+    this.#name = name
+    this.#sub = sub
+    this.#isLocal = isLocal
+  }
+
+  static async create(request: Request) {
+    const user = await getUser(request)
+    const isLocal = new URL(request.url).hostname == 'localhost'
```

Comment:
I see we use the process.env.ARC_ENV in sandboxOidcIdp.js file, would that work here as well? Would be be able to infer it from a testing environment, or does that not work for this case?

Reply:
I use it from time to time. Best not to break it if we can help it. I don't see anything in ARC_ENV that would tell us if we are connected to the real DynamoDB; it has ports.tables set regardless of the value of ARC_ENV.

One option would be to look at the DynamoDB client's endpoint URL.

Have you tried asking about this on the Architect discord?

Reply:
Not yet, i've been reading through the architect docs to try to get a better idea of what is/isn't available already. I didn't see anything that explicitly states it either, but it did get me thinking that maybe it would make sense to just add a variable into our .env? 

I'll ask there shortly though, just to be safe

Reply:
From ryan on discord:

> Assuming this is a Lambda, you can use the ARC_SANDBOX env var to know whether you are in sandbox

I'll explore that option

Reply:
locally, that object looks like this

```
{
  apiType: 'http',
  cwd: 'C:\\Repos\\gcn.nasa.gov',
  ports: { http: 3333, tables: 5555, _arc: 2222 },
  staticPath: 'C:\\Repos\\gcn.nasa.gov\\public',
  version: '5.4.1',
  lambdaSrc: 'C:\\Repos\\gcn.nasa.gov\\server'
}
```

@lpsinger do you know if any of these fields are useful for us to use? I don't know what the values are on the hosted env, but I imagine we could use cwd?

Reply:
Try this...

```ts
(await (db._db as unknown as DynamoDB).config.endpoint?.())?.hostname
```

Reply:
Nice, I updated it to use that